src/views/components/EntityCard.vue

```diff
@@ -0,0 +1,159 @@
+<template>
+	<div class="entity-card" :style="getCardStyle()">
+		<img class="img-tier" 
+			:class="{'user-tier': !this.isOpponent(), 'opponent-tier': this.isOpponent()}" 
+			:src="getEntityTierPath()" />
+
+		<img class="img-entity" 
+			:class="{'user-pet': !this.isOpponent(), 'opponent-pet': this.isOpponent()}" 
+			:src="getEntityPath()" />
+
+		<img class="img-pedestal" src="../../assets/images//layout/shop-pedestal-with-shadow.png" />
+
+		<template v-if="isPet()">
+			<img class="img-stats" src="../../assets/images/layout/animal-stats-template.png" />
+			<span class="card-text text-attack">{{getPetAttack()}}</span>
+			<span class="card-text text-health">{{getPetHealth()}}</span>
+		</template>
+	</div>
+</template>
+
+<script>
+export default {
+  name: 'EntityCard',
+  props: {
+	cardWidth: Number,
+	cardHeight: Number,
+
+	entityType: String, /* Expected values:		"Pet", "Item" */
+	entityName: String, /* Examples: 			"Beaver", "Chocolate" */
+	entityTier: String, /* Examples:			"xp0", "tier3" */
+
+	/* Pet Specific Properties */
+	petAttack: Number,
+	petHealth: Number,
+	petIsOpponent: Boolean,
+  },
+  methods: {
+	isPet: function () {
+		return this.entityType == "Pet";
+	},
+	isItem: function () {
+		return this.entityType == "Item";
+	},
+	isOpponent: function () {
+		return this.petIsOpponent == null ? false : this.petIsOpponent;
+	},
+    getEntityPath: function () {
+	  var subfolder = this.isPet() ? "pets" : "items";
+      return '/src/assets/images/turtle-pack/' + subfolder + '/' + this.entityName + '.webp';
+    },
+	getCardStyle: function () {
+		return 'width:' + this.cardWidth + 'rem;height:' + this.cardHeight + 'rem';
+	},
+	getPetAttack: function () {
+		return this.petAttack;
+	},
+	getPetHealth: function () {
+		return this.petHealth;
+	},
+
+	getEntityTierPath: function () {
+		if(this.entityTier.includes("xp")) {
```

Comment:
this will certainly seem nitpicky but always use single quotes (') to surround literals whenever possible because when you need to nest literals you have to use single quotes on the outside nested part and double quotes on the inside and this makes it easier